Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.py

```diff
@@ -100,6 +97,104 @@ def _oproxy_authorize(self) -> Tuple[str, str, str, str, int]:
                                    f'Oproxy response: {oproxy_response}')
         return access_token, api_url, instance_id, refresh_token, expires_in
 
+    def _get_access_token_with_backoff_strategy(self) -> dict:
+        """ Implements a backoff strategy for retrieving an access token. Logic as follows:
+        - First 60 minutes check for access token once every 1 minute max.
+        - Next 47 hours check for access token once every 10 minute max.
+        - After 48 hours check for access token once every 60 minutes max.
+        Successful fetch of access token resets the counters.
+
+        Returns: The oproxy response or raising a DemistoException
+
+        """
+        err_msg = 'We have found out that your recent attempts to authenticate against the CDL server have failed. ' \
+                  'Therefore we have limited the number of calls that the CDL integration performs. ' \
+                  'If you wish to try authenticating again, please run the `cdl-reset-authentication-timeout` ' \
+                  'command and retry.'
+        integration_context = demisto.getIntegrationContext()
+        first_failure_time = integration_context.get(FIRST_FAILURE_TIME_CONST)
+        last_failure_time = integration_context.get(LAST_FAILURE_TIME_CONST)
+
+        if first_failure_time and last_failure_time:
+            first_failure_datetime = datetime.fromtimestamp(first_failure_time)
+            last_failure_datetime = datetime.fromtimestamp(last_failure_time)
```

Comment:
```suggestion
            last_failure_datetime = datetime.fromisoformat(last_failure_time)
```
Use readable isoformat in the context